### ISSUE TITLE
Speed Limit Widget

### ISSUE THREAD

**Issue by Sonwon1** [osmandapp/OsmAnd#20033]
### Describe the idea (required)

From what I can tell Speed Limit Widget has three states:

1. show the speed limit in the widget.
2. show none if the speed limit is unlimited.
3. If there is no speed limit information then blank the widget and remove it from the screen.

I propose option 3 be changed to when there is no speed limit information replace the speed limit with "- - -" and leave the widget on the screen.

It is distracting having the widget disappear and reappear on the screen which can cause widgets to shift up or down. This just makes it harder to read widgets since they are not at their expected location on the screen.

### Tell us about the expected behaviour (required)

I propose option 3 be changed to when there is no speed limit information replace the speed limit with "- - -" and leave the widget on the screen.

It is distracting having the widget disappear and reappear on the screen which can cause widgets to shift up or down. This just makes it harder to read widgets since they are not at their expected location on the screen.

### Tell us about alternatives you've considered (required)

None

**Comment by pebogufi** [comment]
and 
4. If enforcement=average_speed 
 may be using a background color like yellow ?

**Comment by Sonwon1** [comment]
> and 4. If enforcement=average_speed may be using a background color like yellow ?

Do you mean Actual Speed > Speed Limit then turn background yellow? I think that should be optional along with a settable threshold.

**Comment by pebogufi** [comment]
> > and 4. If enforcement=average_speed may be using a background color like yellow ?
> 
> Do you mean Actual Speed > Speed Limit then turn background yellow? I think that should be optional along with a settable threshold.

No, I think it should be a warning during the complete range of this enforcement, which is normally over some kilometers.
If i drive the whole length with exact the max speed, but at any pos 200 m too fast, i will get a photo.

**Comment by Sonwon1** [comment]
> > > and 4. If enforcement=average_speed may be using a background color like yellow ?
> > 
> > 
> > Do you mean Actual Speed > Speed Limit then turn background yellow? I think that should be optional along with a settable threshold.
> 
> No, I think it should be a warning during the complete range of this enforcement, which is normally over some kilometers. If i drive the whole length with exact the max speed, but at any pos 200 m too fast, i will get a photo.

Okay that is different, then number 4 should be user selected since not everyone will want that.

**Comment by vshcherb** [comment]
Yes we need to support enforcement=average_speed - as it's a specific case

**Comment by TobiPeterG** [comment]
I like the idea of keeping on on the screen, also, would it be possible to actually show the "unlimited speed" sign when the speed limit is "none"?

**Comment by Sonwon1** [comment]
Can I bump this, it has been over a year and this should not be hard to implement.

**Comment by Sonwon1** [comment]
I just noticed there are other screen widgets that disappear when there is no data for them, can we make this an option for all widgets?

**Comment by ejafus** [comment]
I too, like the idea of showing a placeholder when there is no speed limit info. rather than removing the icon.
.
On a related matter, can there be some customization added to the widget so that speed limit caution warnings have different thresholds?  There is a speed limit threshold setting for alerts but it does not seem to affect the widget.  

Currently, the threshold seems to be 5kph above the posted speed limit before the displayed speed I am going turns red.  
This threshold isn't realistic where I live....it's usually more like 10kph at lower speeds and 20kph at higher speeds.
